### PR TITLE
22952 update furnishing job use versioning proxy

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#22952

*Description of changes:*

1. Replaced the code to initialize and use the new db versioning
2. Updated configuration to support feature flags
3. Added flags.json file
4. Updated requirements.txt with necessary dependencies

5. Switch the versioning Proxy
- Flag is on
Screenshot 2024-11-29 at 11 33 39 AM

- Flag is off
Screenshot 2024-11-29 at 11 43 01 AM

6. Unit Test
- legal: True, furnishings: True
Screenshot 2024-11-29 at 11 45 10 AM

- legal: True, furnishings: False
Screenshot 2024-11-29 at 11 46 53 AM

- legal: False, furnishings: True
Screenshot 2024-11-29 at 11 47 48 AM

- legal: False, furnishings: False
Screenshot 2024-11-29 at 11 48 09 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/furnishings/flags.json [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
-        "disable-dissolution-sftp-bclaws": false
+        "disable-dissolution-sftp-bclaws": false,
+        "db-versioning": {
+            "legal-api": true,
+            "emailer": false,
+            "filer": false,
+            "entity-bn": false,
+            "digital-credentials": false,
+            "dissolutions-job": false,
+            "furnishings-job": false,
```
Comment (AimeeGao) [inline jobs/furnishings/flags.json:13]: That would be nice if you could provide some local testing examples to verify if the versioning switches work correctly through the flag of furnishings-job

**Review by AimeeGao — COMMENTED** [review]
Could you provide some local testing examples to verify if the versioning switches work correctly through the flags of

**Review by eason-pan-bc — COMMENTED** [review]
- [x] Please consider adding the summary of changes to your PR, 
- [ ] Please also add some screenshots of local running the furnishing job, and showing it runs as expected when FFs on and off. Thank you.

File: jobs/furnishings/flags.json [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+            "filer": false,
+            "entity-bn": false,
+            "digital-credentials": false,
+            "dissolutions-job": false,
+            "furnishings-job": true,
+            "emailer-reminder-job": false,
+            "future-effective-job": false,
+            "update-colin-filings-job": false,
+            "update-legal-filings-job": false
+        }
```
Comment (kzdev420) [inline jobs/furnishings/flags.json:18]: Add db-versioning flag values

File: jobs/furnishings/src/furnishings/config.py [hunk 3]
```diff
@@ -45,6 +45,8 @@ def get_named_config(config_name: str = 'production'):
 class _Config:  # pylint: disable=too-few-public-methods
     """Base class configuration."""
 
+    # used to identify versioning flag
+    SERVICE_NAME = 'furnishings-job'
```
Comment (kzdev420) [inline jobs/furnishings/src/furnishings/config.py:49]: Add service-name that can identify versioning flag

**Review by AimeeGao — COMMENTED** [review]
Thanks for sharing the screenshot of the unit tests. Could you also share screenshots of locally running the furnishing job as per Eason's suggestion?

we'd like to see:
1. the job running with feature flags turned on
2. the job running with feature flags turned off

That would be a great help, thx.

**Review by eason-pan-bc — APPROVED** [review]
LGTM.
Maybe when you finished working on dissolution jobs ticket, some tests to actually process some businesses in different stages will be very helpful.

**Review by AimeeGao — APPROVED** [review]
LGTM(^_^), thanks for sharing the detailed test screenshots

**Pull request merged**